request-example.go

```diff
@@ -555,6 +547,35 @@ func (fs *root) fetch(path string) (*memFile, error) {
 	return file, nil
 }
 
+// In memory file-system which implements the RealPathFileLister
+type rootWithRealPather struct {
+	root
+}
+
+// implements RealpathFileLister interface
+func (fs *rootWithRealPather) Realpath(p string) (string, error) {
+	if fs.mockErr != nil {
+		return "", fs.mockErr
+	}
+	if fs.startDirectory == "" || fs.startDirectory == "/" {
+		return cleanPath(p), nil
+	}
+	return cleanPathWithBase(fs.startDirectory, p), nil
+}
+
+// In memory file-system which implements legacyRealPathFileLister
+type rootWithLegacyRealPather struct {
+	root
+}
+
+// implements RealpathFileLister interface
+func (fs *rootWithLegacyRealPather) Realpath(p string) string {
+	if fs.startDirectory == "" || fs.startDirectory == "/" {
+		return cleanPath(p)
+	}
+	return cleanPathWithBase(fs.startDirectory, p)
+}
```

Comment:
> Since this `rootWithLegacyRealPather` is not exported, we can have it return some sort of test value. In fact, since it is exported and only used in tests, it should be in a `*_test.go` file.

I just noticed that `root` never implemented the `RealPathFileLister`, the method name was wrong: `Realpath` vs `RealPath`. Maybe we move both `rootWithLegacyRealPather` and `rootWithRealPather` to  `*_test.go` file

Reply:
🙃 yay, missing test case in the original code.